### PR TITLE
Add support for specifying sinkoptions when creating stackdriver sink

### PR DESCRIPTION
**Pull request by gargnupur**

### CODE REVIEW

File: mixer/adapter/stackdriver/config/config.proto [hunk 1]
```diff
@@ -73,7 +73,9 @@ message Params {
     // Describes how to represent an Istio metric in Stackdriver.
     // See https://github.com/googleapis/googleapis/blob/master/google/api/metric.proto
     message MetricInfo {
+        // The kind of measurement for metric. It describes how the data is reported. Ex: Gauge.
```
Comment (geeknoid) [inline mixer/adapter/stackdriver/config/config.proto]: for metric. It describes -> for a metric, which describes

File: mixer/adapter/stackdriver/config/config.proto [hunk 2]
```diff
@@ -73,7 +73,9 @@ message Params {
...
         google.api.MetricDescriptor.MetricKind kind = 1;
+        // The value type of a metric. Ex: Distribution.
```
Comment (geeknoid) [inline mixer/adapter/stackdriver/config/config.proto]: The type of the metric's value.

File: mixer/adapter/stackdriver/config/config.proto [hunk 3]
```diff
@@ -222,6 +224,26 @@ message Params {
...
+            // be a unique service account used only for exports from the new sink.
+            bool UniqueWriterIdentity = 4;
+
+            // These fields apply only to UpdateSinkOpt calls. The corresponding sink field
+            // is updated if and only if the Update field is true.
+            // Upate sink destination.
+            bool UpdateDestination = 5;
+            // Update sink filter.
+            bool UpdateFilter = 6;
+            // Update inlcude children.
```
Comment (geeknoid) [inline mixer/adapter/stackdriver/config/config.proto]: inlcude -> includes

File: mixer/adapter/stackdriver/config/config.proto (hunk 1 above)
Comment (gargnupur) [reply mixer/adapter/stackdriver/config/config.proto]: Fixed

File: mixer/adapter/stackdriver/config/config.proto (hunk 2 above)
Comment (gargnupur) [reply mixer/adapter/stackdriver/config/config.proto]: Fixed

File: mixer/adapter/stackdriver/config/config.proto (hunk 3 above)
Comment (gargnupur) [reply mixer/adapter/stackdriver/config/config.proto]: Fixed

**Comment by gargnupur** [comment]
Closing this pull request as this is based out of master and re-basing it to 1.0 was pulling in dependencies from master to 1.0.
Created new pr for 1.0 -> https://github.com/istio/istio/pull/7944

**Comment by gargnupur** [comment]
Reopening as there needs to be a separate PR for master too as per https://github.com/istio/istio/issues/7672#issuecomment-413415946

**Pull request closed without merging**